app/src/main/java/com/github/se/project/ui/lesson/AddLesson.kt

```diff
@@ -0,0 +1,239 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.android.sample.model.lesson.Lesson
+import com.android.sample.model.lesson.LessonStatus
+import com.android.sample.model.lesson.LessonsViewModel
+import com.github.se.project.model.profile.Language
+import com.github.se.project.model.profile.ListProfilesViewModel
+import com.github.se.project.model.profile.TutoringSubject
+import com.github.se.project.ui.components.WritableDropdown
+import com.github.se.project.ui.navigation.NavigationActions
+import com.github.se.project.ui.navigation.Screen
+import java.util.Calendar
+
+@Composable
+fun AddLessonScreen(
+    navigationActions: NavigationActions,
+    listProfilesViewModel: ListProfilesViewModel =
+        viewModel(factory = ListProfilesViewModel.Factory),
+    lessonViewModel: LessonsViewModel = viewModel(factory = LessonsViewModel.Factory)
+) {
+  var title by remember { mutableStateOf("") }
+  var description by remember { mutableStateOf("") }
+  var subject by remember { mutableStateOf("") }
+  var language by remember { mutableStateOf("") }
+  var minPrice by remember { mutableStateOf(0.0) }
+  var maxPrice by remember { mutableStateOf(0.0) }
+  var timeSlot by remember { mutableStateOf("") }
+  val calendar = Calendar.getInstance()
+
+  val profile = listProfilesViewModel.currentProfile.collectAsState()
+  val lessonState = lessonViewModel.lessons
+
+  // Context for the Toast messages
+  val context = LocalContext.current
+
+  var selectedDate by remember { mutableStateOf("") }
+  var selectedTime by remember { mutableStateOf("") }
+
+  val onConfirm = {
+    val error =
+        validateLessonInput(
+            title,
+            description,
+            subject,
+            language,
+            minPrice,
+            maxPrice,
+            selectedDate,
+            selectedTime,
+        )
+    if (error != null) {
+      Toast.makeText(context, error, Toast.LENGTH_SHORT).show()
+    } else {
+      lessonViewModel.addLesson(
+          profile.value!!.uid,
+          Lesson(
+              lessonViewModel.getNewUid(),
+              title,
+              description,
+              TutoringSubject.valueOf(subject),
+              "",
+              profile.value!!.uid,
+              minPrice,
+              maxPrice,
+              "${selectedDate}T${selectedTime}:00",
+              LessonStatus.REQUESTED,
+              language))
+
+      // this should ideally be done in onSuccess callback of addLesson
+      // but done directly here in the meantime
+      navigationActions.navigateTo(Screen.HOME)
+    }
+  }
+
+  // Date Picker
+  val datePickerDialog =
+      DatePickerDialog(
```

Comment:
I noticed while testing the feature that we can select past dates, which does not make sense for our app. You can restrict the selection of date with this line of code :)
    datePickerDialog.datePicker.minDate = System.currentTimeMillis()

Reply:
Done